packages/split-view/src/SplitView.ts

```diff
@@ -0,0 +1,416 @@
+*/
+
+import {
+    html,
+    SpectrumElement,
+    CSSResultArray,
+    TemplateResult,
+    property,
+    css,
+    PropertyValues,
+    ifDefined,
+} from '@spectrum-web-components/base';
+
+import styles from './split-view.css.js';
+
+const ORIENTATIONS = {
+    horizontal: 'width',
+    vertical: 'height',
+};
+
+const CURSORS = {
+    horizontal: {
+        default: 'ew-resize',
+        min: 'e-resize',
+        max: 'w-resize',
+    },
+    vertical: {
+        default: 'ns-resize',
+        min: 's-resize',
+        max: 'n-resize',
+    },
+};
+
+const COLLAPSE_THREASHOLD = 50;
+
+const SPLITTERSIZE = 2;
+
+/**
+ * @element sp-split-view
+ */
+export class SplitView extends SpectrumElement {
+    public static get styles(): CSSResultArray {
+        return [
+            styles,
+            css`
+                .pane {
+                    overflow: hidden;
+                }
+                .secondaryPane {
+                    flex: 1;
+                }
+                #splitter {
+                    height: auto; /* for horizontal splitviews without proper outter height value */
+                }
+            `,
+        ];
+    }
+
+    @property({ type: Boolean, reflect: true })
+    public vertical = false;
+
+    @property({ type: Boolean, reflect: true })
+    public resizable = false;
+
+    @property({ type: Boolean, reflect: true })
+    public collapsible = false;
+
+    /** The minimum size of the primary pane */
+    @property({ type: Number, attribute: 'primary-min' })
+    public primaryMin = 304;
+
+    /** The maximum size of the primary pane */
+    @property({ type: Number, attribute: 'primary-max' })
+    public primaryMax = Infinity;
+
+    /** The size of the primary pane */
+    @property({ type: Number, attribute: 'primary-size' })
+    public primarySize?: number;
+
+    /** The default size of the primary pane */
+    @property({ type: Number, attribute: 'primary-default' })
+    public primaryDefault = 304;
+
+    /** The minimum size of the secondary pane */
+    @property({ type: Number, attribute: 'secondary-min' })
+    public secondaryMin = 304;
+
+    /** The maximum size of the secondary pane */
+    @property({ type: Number, attribute: 'secondary-max' })
+    public secondaryMax = Infinity;
+
+    @property({ type: Number, attribute: false })
+    public dividerPosition = this.primaryDefault;
+
+    @property({ type: Number, attribute: false })
+    public minPos = 0;
+
+    @property({ type: Number, attribute: false })
+    public maxPos = Infinity;
+
+    @property({ type: Boolean, reflect: true, attribute: 'is-collapsed-start' })
+    public isCollapsedStart = this.dividerPosition === 0;
+
+    @property({ type: Boolean, reflect: true, attribute: 'is-collapsed-end' })
+    public isCollapsedEnd = false;
+
+    @property({ type: Boolean, reflect: true })
+    public dragging = false;
+
+    @property({ type: Boolean, reflect: true })
+    public hovered = false;
```

Comment:
I'm not so happy with the way the upstream implementation forces the use of more complex JS to handle the styling of this. It's an internal implementation, however, so we can refactor this at a later date if you're comfortable with this approach.

Leveraging the matched `:hover` styles on `.spectrum-SplitView-splitter` we could do some special magic with:
```
:host(:not([vertical])) #splitter {
  padding: 0 10px; // this way we could also use Custom Properties to allow this math to be managed by the Scaling system
  margin: 0 -10px;
}
```
To expand the addressable area of the splitter, allowing CSS to chose when "hover" has occurred. This would mean we'd only need to manage what cursor to show, not whether to show it.

Reply:
I tried so, but could not successfully use this pattern as due to the bg-color the splitter gets 10px width colored. Maybe we can come up with another solution later.

Reply:
Sorry, I forgot the most important part of the CSS above:
```
background-clip: content-box;
```
Then the background color should no longer grow that way 🤞🏼

Reply:
it works!